base/operators.jl

```diff
@@ -856,6 +860,15 @@ julia> ∘(fs...)(3)
 3.0
 ```
 """
+function ∘()
+    # Like +() and *() we leave ∘() undefined.
+    # While `∘() = identity` is a reasonable definition for functions, this
+    # would cause headaches for composition of user defined morphisms.
+    # See also #34251
+    msg = """Empty composition ∘() is undefined."""
```

Comment:
Why define it at all then? Why not just let it be a method error?

Reply:
So that we don't get a stream of PRs that want to define it :smile:

Reply:
I think it's better to leave it undefined.

Reply:
I see the benefit of making it very clear that not defining it is intentional with this approach.  But I think leaving a comment is enough to prevent future PRs.